crates/auth/src/types.rs

```diff
@@ -0,0 +1,166 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
```

Comment:
How about we extract a `user.rs` containing the `User` struct and all the functionality related?
Then we could also have a `providers.rs` containing functionality related to different providers (only GitHub for now)